pmd-java/src/main/java/net/sourceforge/pmd/lang/java/qname/QualifiedNameFactory.java

```diff
@@ -105,21 +105,16 @@ public static JavaTypeQualifiedName ofClass(Class<?> clazz) {
             return null;
         }
 
-        String name = clazz.getName();
-        if (name.indexOf('.') < 0) {
-            name = '.' + name; // unnamed package, marked by a full stop. See ofString's format below
-        }
-
-        // Note: this assumes, that clazz has been loaded through the correct classloader,
-        // specifically through the auxclasspath classloader.
-        // But this method should only be used in tests anyway
-        return (JavaTypeQualifiedName) ofStringWithClassLoader(name, clazz.getClassLoader());
+        // We preserve the known type until within JTypeQualifiedName. It won't call a ClassLoader
+        return (JavaTypeQualifiedName) ofString(clazz.getName(), clazz, null);
     }
 
 
     /**
-     * Parses a qualified name given in the format defined for this implementation. The format
-     * is specified by a regex pattern (see {@link #FORMAT}). Examples:
+     * Parses a qualified name given in the format defined for this implementation.
+     * This uses a default classloader, prefer {@link #ofString(String, ClassLoader)}.
```

Comment:
shall this method even exist if we now it will provide subpar results due to not using the auxclasspath classloader?

Reply:
Actually JavaTypeQualifiedName is fundamentally flawed, in that that representation expects to somehow *know* which segments represent an inner class and which represent a package name. This is ok for the use we have of it today, but there's no way we can use it for JResolvableClassSymbol, because at the source level package and type names are both separated with the same delimiter (a dot `.`). So unless we load the class eagerly, we can't satisfy the contract of JavaTypeQualifiedName (and will get inconsistent/wrong results), and if we do load the class eagerly, then we completely defeat the purpose. So the best we can do is use strings anyway, and make a symbol available when we have the info.

Fortunately caching metrics on nodes allows us to completely scrap the clumsy metrics cache (10,000 lines IIRC), and so the qualified name class too. I should document that on the wiki, but in the meantime, I wouldn't be too concerned about this class.